metrics_ci_integration/lib/jenkins/client/jenkins_client.dart

```diff
@@ -0,0 +1,278 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:ci_integration/common/authorization/authorization.dart';
+import 'package:ci_integration/jenkins/client/config/tree_query.dart';
+import 'package:ci_integration/jenkins/client/model/jenkins_build.dart';
+import 'package:ci_integration/jenkins/client/model/jenkins_build_artifact.dart';
+import 'package:ci_integration/jenkins/client/model/jenkins_building_job.dart';
+import 'package:ci_integration/jenkins/client/model/jenkins_job.dart';
+import 'package:ci_integration/jenkins/client/model/jenkins_multi_branch_job.dart';
+import 'package:ci_integration/jenkins/client/model/jenkins_query_limits.dart';
+import 'package:ci_integration/jenkins/client/model/jenkins_result.dart';
+import 'package:ci_integration/jenkins/client/util/url_utils.dart';
+import 'package:http/http.dart';
+import 'package:meta/meta.dart';
+
+/// A callback for parsing Jenkins API response body.
+typedef BodyParserCallback<T> = JenkinsResult<T> Function(Map<String, dynamic>);
```

Comment:
```suggestion
typedef BodyParserCallback<T> = JenkinsResult<T> Function(Map<String, dynamic> body);
```
Can we add the parameter name to be able to generate valid suggestions in IDE? 
I mean that if you'll use suggestion and generate `_handleResponse` call you'll obtain such line of code: 
`_handleResponse(responseFuture, () => null)` which is not valid because closure should take a param.